### PR TITLE
Mount the audit config dir for kube-apiserver

### PR DESCRIPTION
**Pull request by hakman**

### CODE REVIEW

File: pkg/apis/kops/validation/validation.go [hunk 1]
```diff
@@ -671,6 +672,17 @@ func validateKubeAPIServer(v *kops.KubeAPIServerConfig, c *kops.Cluster, fldPath
...
 	}
 
+	if v.AuditPolicyFile == "" && v.AuditWebhookConfigFile != "" {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("auditWebhookConfigFile"), v.AuditWebhookConfigFile, "an audit policy is required for the audit webhook config"))
+	}
+	if v.AuditPolicyFile != "" && v.AuditWebhookConfigFile != "" {
+		auditPolicyDir := filepath.Dir(v.AuditPolicyFile)
+		auditWebhookConfigDir := filepath.Dir(v.AuditWebhookConfigFile)
+		if auditPolicyDir != auditWebhookConfigDir {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("auditWebhookConfigFile"), v.AuditWebhookConfigFile, "the audit webhook config must be placed in the same dir as the audit policy"))
```
Comment (johngmyers) [inline pkg/apis/kops/validation/validation.go:682]:
```suggestion
			allErrs = append(allErrs, field.Invalid(fldPath.Child("auditWebhookConfigFile"), v.AuditWebhookConfigFile, "the audit webhook config must be placed in the same directory as the audit policy"))
```

**Review by johngmyers — APPROVED** [review]

**Comment by math3vz** [comment]
@hakman @johngmyers it looks like this PR can introduce a breaking change if `auditPolicyFile` is placed at `/srv/kubernetes`, it causes two volume mounts to be added with the same value and the API server will not start.

The error:
`Nov 27 14:50:38 ip-10-34-32-105 kubelet[7165]: E1127 14:50:38.349932    7165 file.go:187] "Could not process manifest file" err="invalid pod: [spec.containers[0].volumeMounts[15].mountPath: Invalid value: \"/srv/kubernetes/kube-apiserver\": must be unique]" path="/etc/kubernetes/manifests/kube-apiserver.manifest"`

`/etc/kubernetes/manifests/kube-apiserver.manifest`

```
    - mountPath: /srv/kubernetes/kube-apiserver
      name: srvkapi
      readOnly: true
    - mountPath: /srv/sshproxy
      name: srvsshproxy
      readOnly: true
    - mountPath: /var/log
      name: auditlogpathdir
    - mountPath: /srv/kubernetes/kube-apiserver
      name: auditconfigdir
      readOnly: true
```

**Comment by hakman** [comment]
@math3vz I think this was fixed in later versions of kOps. Could you check please?

**Comment by rsicart** [comment]
> @hakman @johngmyers it looks like this PR can introduce a breaking change if `auditPolicyFile` is placed at `/srv/kubernetes`, it causes two volume mounts to be added with the same value and the API server will not start.
> 
> The error: `Nov 27 14:50:38 ip-10-34-32-105 kubelet[7165]: E1127 14:50:38.349932 7165 file.go:187] "Could not process manifest file" err="invalid pod: [spec.containers[0].volumeMounts[15].mountPath: Invalid value: \"/srv/kubernetes/kube-apiserver\": must be unique]" path="/etc/kubernetes/manifests/kube-apiserver.manifest"`
> 
> `/etc/kubernetes/manifests/kube-apiserver.manifest`
> 
> ```
>     - mountPath: /srv/kubernetes/kube-apiserver
>       name: srvkapi
>       readOnly: true
>     - mountPath: /srv/sshproxy
>       name: srvsshproxy
>       readOnly: true
>     - mountPath: /var/log
>       name: auditlogpathdir
>     - mountPath: /srv/kubernetes/kube-apiserver
>       name: auditconfigdir
>       readOnly: true
> ```

Yesterday I had exactly this same error when upgrading from 1.25 to 1.26, using kops 1.26.6:

```
kubelet[2803]: E1205 14:27:57.582433    2803 file.go:187] "Could not process manifest file" err="invalid pod: [spec.containers[0].volumeMounts[15].mountPath: Invalid value: \"/srv/kubernetes/kube-apiserver\": must be unique]" path="/etc/kubernetes/manifests/kube-apiserver.manifest"
```

Updating the cluster manifest with values proposed in the commit fixed the issue, allowing apiserver to start. I replaced `/srv/kubernetes/kube-apiserver/audit-policy-config.yaml` by `/etc/kubernetes/audit/policy-config.yaml`.

**Pull request merged**